src/printer.js

```diff
@@ -3014,9 +3014,11 @@ function printJSXChildren(path, options, print, jsxWhitespace) {
           children.push(hardline);
         }
       } else if (/\s/.test(value)) {
-        // whitespace-only without newlines,
-        // eg; a single space separating two elements
-        children.push(jsxWhitespace);
+        // whitespace(s)-only without newlines,
```

Comment:
@vjeux here comes the fix, the previous implementation was not taking multiple consecutive spaces into account.